### ISSUE TITLE
Can't find any of the go tools?

### ISSUE THREAD

**Issue by polds** [Microsoft/vscode-go#10]
I've been trying to install the plugin and not having any success, every time I start writing anything I see notifications like:

```
The 'gocode' command is not available. Use 'go get -u github.com/nsf/gocode' to install.
```

and all the other tools it needs too. Even those these tools are in fact installed.

I've verified $GOPATH/bin is in my $PATH and I'm running on Go 1.5.1 installed with brew.

**Comment by lukehoban** [comment]
What do you see when you run the `Go: Current GOPATH` command from the command palette (`cmd-shift-p`)?

If it's not what you expect, you can override the GOPATH to use by adding the "go.gopath" setting.  See https://github.com/Microsoft/vscode-go#options.

**Comment by Hokutosei** [comment]
This is running me insane, I am sure i have all this tools, as I am using it in atom. but in VS, it keeps getting me this same error. how do i fix? 
followed instructions from here https://marketplace.visualstudio.com/items/lukehoban.Go

**Comment by Hokutosei** [comment]
from `settings.json` 

```
    "go.gopath": {
        "type": "string",
        "default": null,
        "description": "Specifies the GOPATH to use when no environment variable is set."
    }
```

**Comment by jchannon** [comment]
Is this the same as #11 ?

**Comment by polds** [comment]
D'oh. I've figured out my issue. I had literally copy + pasted what the instructions had:

```
{
    "go.buildOnSave": {
        "type": "boolean",
        "default": true,
        "description": "Run 'go build'/'go test' on save."
    },
    "go.lintOnSave": {
        "type": "boolean",
        "default": true,
        "description": "Run 'golint' on save."
    },
    "go.vetOnSave": {
        "type": "boolean",
        "default": true,
        "description": "Run 'go tool vet' on save."
    },
    "go.formatTool": {
        "type": "string",
        "default": "goreturns",
        "description": "Pick 'gofmt', 'goimports' or 'goreturns' to run on format."
    },
    "go.gopath": {
        "type": "string",
        "default": null,
        "description": "Specifies the GOPATH to use when no environment variable is set."
    }
}
```

Those are just the available options! Really your settings should look like this: 

```
{
    "go.formatTool": "gofmt",
    "go.lintOnSave": true,
    "go.gopath": "/Users/peter/dev/go"
}
```

**Comment by lukehoban** [comment]
@jchannon Yes, I think these are the same issue.

For those you seeing this problem - could you share what you see when you run the `Go: Current GOPATH` command from the command palette (`cmd-shift-p`)?

**Comment by lukehoban** [comment]
@polds Good point, I'll update the README to make it more clear how to use those settings.

**Comment by polds** [comment]
Re: The comment above if your "Current GOPATH" shows `Current GOPATH:[object Object]` that's a hint your settings.json is incorrect.

**Comment by jchannon** [comment]
My GOPATH is as expected `/Users/jonathan/Projects/Go` but `/Users/jonathan/Projects/Go/bin` is in my PATH

**Comment by lukehoban** [comment]
It shouldn't matter what is in your PATH, the tools are located relative to your GOPATH.

If you launch Code from a shell with a GOPATH set, you should see that running `Go: Current GOPATH` shows the correct value and tools should be located there.

You can also override any other GOPATH settings by adding the `"go.gopath"` setting to your settings.json (cmd+,):

```json
{   
    "go.gopath": "/Users/lukeh/go"
}
```

If anyone is having problems after these steps though, could you share details about your environment?

**Comment by Hokutosei** [comment]
setting it for an hour, but no luck. here is my `go env`

```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/jeanepaul/Developer/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GO15VENDOREXPERIMENT=""
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fno-common"
CXX="clang++"
CGO_ENABLED="1"
```

**Comment by lukehoban** [comment]
@Hokutosei What do you see when you run `Go: Current GOPATH` in VS Code?  Did overriding GOPATH with the `go.gopath` setting as mentioned earlier in the thread work for you?

**Comment by jchannon** [comment]
How do you launch it from shell?

**Comment by Hokutosei** [comment]
@lukehoban 
```
// Place your settings in this file to overwrite the default settings
{
    "go.buildOnSave": true,
    "go.lintOnSave": true,
    "go.vetOnSave": true,
    "go.formatTool": "go fmt",
    "go.gopath": "/Users/jeanepaul/Developer/go"
}
```

after browser restart, `Current GOPATH:/Users/jeanepaul/Developer/go` hold on, i'm seeing light

**Comment by Gys** [comment]
I normally use Atom and that works fine. I also got the 'missing tools' message. Would be good if it tells which tool instead of offering to install them, because I like to know what is happening. Anyway, I checked the list and the one tool that VS Code by default needs in addition to the Atom setup is

    go-find-references (go get -u -v github.com/lukehoban/go-find-references).

Also I found that VS Code has to be started from the commandline. Otherwise the message keeps coming up. This is also in the readme but maybe somebody missed it.
So I do this:

    /Applications/Visual\ Studio\ Code.app/Contents/MacOS/Electron

Still, the first time the message appeared but after clicking it the editor was satisfied after all.

**Comment by Hokutosei** [comment]
@jchannon 
in `~/.zshrc` just add code below

```
# MS visual code
code () { VSCODE_CWD="$PWD" open -n -b "com.microsoft.VSCode" --args $* ;}
```

**Comment by jchannon** [comment]
@Hokutosei thanks that solved it plus I now get the `find-references` notification that @Gys mentoned and I think we're now all good

**Comment by Hokutosei** [comment]
nope, its not working for me. edit some structs, nope, indents are not fixed when `cmd+r` unlike in atom.  oh no, spent almost 2hrs with this and its 3am. gtg

**Comment by lukehoban** [comment]
@Hokutosei sorry about that - if you do have time to share more details of your environment sometime that would be great.

**Comment by mgutz** [comment]
@lukehoban Shouldn't tools be searched in `$GOROOT/bin`? `$GOPATH` can be multiple paths like `/Users/mgutz/go:/User/mgutz/workspace1`. We download the Go tarballs and set `$GOROOT` explicitly. The Go tools will set GOROOT internally. `go get` installs to `$GOROOT/bin` if I'm not mistaken.

**Comment by shurcooL** [comment]
From [Go documentation](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable):

> The GOPATH environment variable lists places to look for Go code. On Unix, the value is a colon-separated string. On Windows, the value is a semicolon-separated string. On Plan 9, the value is a list.

Is this extension implementing that correctly?

My GOPATH is a colon-separated string that includes a couple workspaces. Is that the reason it's not able to find any of the Go tools?

I'm not sure if it's related because I'd imagine it should use PATH to find binaries. My PATH also contains multiple entries.

**Comment by Gys** [comment]
$GOROOT/bin is not the only place. External tools like goreturns and golint are installed with 'go get' and therefore are compiled to $GOPATH/bin

**Comment by lukehoban** [comment]
@shurcooL Looks like there was a bug in the logic for finding binaries when there are multiple workspaces.  That should be fixed with b1e2a93e8d7883f2f03f9e206ea720178235d8c5.  I pushed version 0.6.7 with that fix so if you update the extension you can see if that fixes this for you.  There is still an issue where you will see "Analysis Tools Missing" if tools are not installed in the first of your workspaces, I'll open a separate issue on that.

**Comment by shurcooL** [comment]
I'm trying to see if 0.6.7 makes a difference. The way I was testing it previously is opening a .go file, adding some unneeded whitespace and seeing if it gets formatted (via `gofmt`) on save. It wasn't.

However, I've realized now that it seemingly not a supported feature. There are these options for on-save functionality:

-   go.buildOnSave
-    go.lintOnSave
-    go.vetOnSave

But no go.formatOnSave, is there?

So, I've tried right clicking and doing "Format Code", and that appears to work now (although I am seeing some strange behavior regarding newlines, perhaps it's to do with `\n` vs `\r\n`?). I haven't tried that with 0.6.6, but I am guessing it wouldn't have worked before, so your fix did fix that. Thank you!

**Comment by Hokutosei** [comment]
Greetings,

@lukehoban  thank you for your efforts, everything seems to be working, those popping errors were gone, but still I need `go fmt` on save.. will be watching this close, want to use other IDE. and vscode looks beautiful, and fast. thanks again :)

**Comment by amff** [comment]
I am not versed in typescript nor an experienced github user to submit a pull request but,
from my debugging session (I really need support for multiple paths in gopath) it seems like the function getBinPath() in goPath.ts is searching the filesystem for executables without the .exe extension.

Should be a quick fix.

@lukehoban thank you for this great work

**Comment by lukehoban** [comment]
There are some improvements in 0f20bc67df7fa0abd7df7c75b5d36235a07fac94 and 72ef57ac672ef2deacee7426dce938b0b5bf2f63 that should help with this.

If you upgrade to `0.6.10` you should no longer see most of the problems raised in this issue.

I'll close this for now, but if there are any specific problems remaining please re-open this or another issue.

**Comment by hmny** [comment]
still have the same problem with v. 0.6.19 on Windows 10

**Comment by zanderz** [comment]
I have this. First time trying VS Code, all excited at first but... have to go back to liteide for now :disappointed:  
v 0.10.8 on Windows 10 x64
`The "godef" command is not available. Use "go get -u github.com/rogpeppe/godef" to install.`
`Current GOPATH:c:\work`
godef.exe is installed in `c:\work\bin`

**Comment by Johnaka** [comment]
hey folks are you still getting this issue ? the issue is not in the extension. just set the GOBIN path and run go install  "main.go"  and set the git path on the sys variable environment

**Comment by tsirolnik** [comment]
Hi, I've upgrade vscode from 1.0.0 (which had no problem) to 1.2 and now I have this issue

**Comment by ericmdantas** [comment]
Same here. I'm using 1.2.1.

**Comment by biessek** [comment]
In windows if you set the variable "PATH" with:  
`
set GOPATH="C:\work"
`  
Try without the quotes:  
`
set GOPATH=C:\work
`  
To me, that makes it work.

**Comment by superzadeh** [comment]
@biessek this also worked for me. Go figure...

**Comment by b1ackb0x** [comment]
I just started with Go and had this issue on latest version on windows 10. Turned out, I had to add the workspace bin path in the PATH environment variable. Works fine now.

**Comment by xobs** [comment]
As others have stated, on Windows 8.1, vscode 1.4.0 would repeatedly ask to install plugins, and would even do so when clicking "Install All".

The only solution that allowed it to work was to add %GOPATH%\bin to my PATH and restart vscode.  Without adding it to the PATH it was able to install dependencies but could not run them, and therefore did not think they were actually installed.

**Comment by ramya-rao-a** [comment]
@xobs @b1ackb0x @superzadeh @biessek 

  https://github.com/Microsoft/vscode-go/pull/457 has fixed this issue and the fix will be out in the next update of the extension.
You shouldn't have to put %GOPATH%\bin to your PATH after the fix

**Comment by akman003** [comment]
I ran into a similar issue. Confirmed my GOPATH was set correctly via `Go: Current GOPATH` in the command palette, but every time I saved a file I'd get prompted to install a few tools I supposedly didn't have. After a few frustrating minutes I restarted VS Code and all was well.

**Comment by ramya-rao-a** [comment]
This fix is now available in the latest version of the Go extension : 0.6.44
No need to restart VS Code anymore

**Comment by maddin2016** [comment]
For me problem was that my machine is behind a proxy. After set `http_proxy` variable all is working fine.

**Comment by alecha** [comment]
Unfortunately I've the same problem, latest version of code and extension. My GOPATH is ok, I've also tried to set it in the `settings.json` file

**Comment by ramya-rao-a** [comment]
@alecha Can you elaborate on what issue are you facing?

**Comment by alecha** [comment]
the same of other users, my gopath is ok but when I save (so when the build runs) I get the error `Error: Error: spawn EACCES`

This is my settings.json file

```json
{
    "go.buildOnSave": true,
    "go.lintOnSave": false,
    "go.vetOnSave": false,
    "go.formatTool": "goimports",
    "go.gopath": "MyGoPath"
}
```

**Comment by ramya-rao-a** [comment]
@alecha What do you see when you run "Go: Current GOPATH" in the command palette?

This current thread is about the extension not able to find the installed tools which I believe is not your case. 

Can you try any of the other features? Are they erroring out too?

- Ctrl+Shift+O for outline of your file
- Right-click -> Format document
- Rename a variable

When you see the `Errpr: spawn EACCES`, what do you see in the console in the developer tools? You can open dev tools by `Developer: Toggle Developer Tools`

Also what version of VS Code and the Go extension are you using and which OS?

**Comment by alecha** [comment]
GoPath is ok: `Current GOPATH:/Users/alessiocavaleri/Documents/Code/go/`
MacOS 10.11.6, Code 1.6.1, extension 0.6.47

- renaming a variable does not throw an error
- format the document does not throw an error

the error is 

```
EACCES: Error: spawn EACCES
    at exports._errnoException (util.js:1026:11)
    at ChildProcess.spawn (internal/child_process.js:313:11)
    at exports.spawn (child_process.js:392:9)
    at Object.exports.execFile (child_process.js:155:15)
    at Object.module.(anonymous function) [as execFile] (ELECTRON_ASAR.js:195:20)
    at /Users/alessiocavaleri/.vscode/extensions/lukehoban.Go-0.6.47/out/src/util.js:113:12
    at Object.getGoVersion (/Users/alessiocavaleri/.vscode/extensions/lukehoban.Go-0.6.47/out/src/util.js:112:12)
    at Object.setupGoPathAndOfferToInstallTools (/Users/alessiocavaleri/.vscode/extensions/lukehoban.Go-0.6.47/out/src/goInstallTools.js:154:12)
    at activate (/Users/alessiocavaleri/.vscode/extensions/lukehoban.Go-0.6.47/out/src/goMain.js:43:22)
    at Function.t._callActivateOptional (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:13:15169)
```

**Comment by ramya-rao-a** [comment]
@alecha 
Looks like the problem is when the extension runs "go" commands like `go build`, `go vet`.
This may be related to #585

Can you try running any of the unit tests from the command palette? Since the extension runs tests via `go test` you should see the same error.

**Comment by EGJ1996** [comment]
Still having the same issue with the following settings:

{
    "window.zoomLevel": 0,
    "go.buildOnSave": true,
    "go.lintOnSave": true,
    "go.vetOnSave": true,
    "go.buildTags": "",
    "go.buildFlags": [],
    "go.lintTool": "golint",
    "go.lintFlags": [],
    "go.vetFlags": [],
    "go.testOnSave": false,
    "go.coverOnSave": false,
    "go.useCodeSnippetsOnFunctionSuggest": false,
    "go.formatOnSave": true, 
    "go.formatTool": "goreturns",
    "go.formatFlags": [],
    "go.goroot": "C:/Users/Dell/Desktop/Go Compiler",
    "go.gopath": "C:/Users/Dell/Desktop/Go Compiler/Chap1",
    "go.inferGopath": false,
    "go.gocodeAutoBuild": false
}

**Comment by ramya-rao-a** [comment]
@EGJ1996 Can you elaborate on what issue you are seeing?

**Comment by EGJ1996** [comment]
@ramya-rao-a  Thanks for the response! 

Error: spawn C:\Users\Dell\Desktop\Go Compiler\bin\go.exe ENOENT
gorename:
Error: spawn C:\Users\Dell\Desktop\Go Compiler\bin\go.exe ENOENT
godef:
Error: spawn C:\Users\Dell\Desktop\Go Compiler\bin\go.exe ENOENT
goreturns:
Error: spawn C:\Users\Dell\Desktop\Go Compiler\bin\go.exe ENOENT
golint:
Error: spawn C:\Users\Dell\Desktop\Go Compiler\bin\go.exe ENOENT
gotests:
Error: spawn C:\Users\Dell\Desktop\Go Compiler\bin\go.exe ENOENT

**Comment by ramya-rao-a** [comment]
@EGJ1996 And you are sure that go.exe exists in "C:\Users\Dell\Desktop\Go Compiler\bin" ?

**Comment by EGJ1996** [comment]
I fixed that issue but now I am still getting these popups:

![capture](https://cloud.githubusercontent.com/assets/12373418/23636851/3b80bd40-02d8-11e7-9b5b-4d6e23e2cd5c.PNG)

**Comment by ramya-rao-a** [comment]
@EGJ1996 Is this the first time you are using the Go extension? Those pop ups appear because the extension cannot find the Go tools in your GOPATH. 

If you are doing this for the first time, then you need to install the Go tools as per the pop ups.

If you have already installed these Go tools, then then find where you installed them and add the corresponding GOPATH in `go.toolsGopath` setting

**Comment by amazingandyyy** [comment]
I keep getting this error message. (MacOS VSCode)

![image](http://i.imgur.com/zZ4JsP0.png)

and after I choose Install All

It only show 

```
Installing 9 tools
  gocode
  gopkgs
  go-outline
  go-symbols
  guru
  gorename
  godef
  golint
  gotests

Installing gocode FAILED
Installing gopkgs FAILED
Installing go-outline FAILED
Installing go-symbols FAILED
Installing guru FAILED
Installing gorename FAILED
Installing godef FAILED
Installing golint FAILED
Installing gotests FAILED

9 tools failed to install.

gocode:
Error: Command failed: /usr/local/bin/go get -u -v github.com/nsf/gocode
package github.com/nsf/gocode: cannot download, $GOPATH not set. For more details see: go help gopath
package github.com/nsf/gocode: cannot download, $GOPATH not set. For more details see: go help gopath

gopkgs:
Error: Command failed: /usr/local/bin/go get -u -v github.com/tpng/gopkgs
package github.com/tpng/gopkgs: cannot download, $GOPATH not set. For more details see: go help gopath
package github.com/tpng/gopkgs: cannot download, $GOPATH not set. For more details see: go help gopath

go-outline:
Error: Command failed: /usr/local/bin/go get -u -v github.com/lukehoban/go-outline
package github.com/lukehoban/go-outline: cannot download, $GOPATH not set. For more details see: go help gopath
package github.com/lukehoban/go-outline: cannot download, $GOPATH not set. For more details see: go help gopath

go-symbols:
Error: Command failed: /usr/local/bin/go get -u -v github.com/newhook/go-symbols
package github.com/newhook/go-symbols: cannot download, $GOPATH not set. For more details see: go help gopath
package github.com/newhook/go-symbols: cannot download, $GOPATH not set. For more details see: go help gopath

guru:
Error: Command failed: /usr/local/bin/go get -u -v golang.org/x/tools/cmd/guru
Fetching https://golang.org/x/tools/cmd/guru?go-get=1
Parsing meta tags from https://golang.org/x/tools/cmd/guru?go-get=1 (status code 200)
get "golang.org/x/tools/cmd/guru": found meta tag main.metaImport{Prefix:"golang.org/x/tools", VCS:"git", RepoRoot:"https://go.googlesource.com/tools"} at https://golang.org/x/tools/cmd/guru?go-get=1
get "golang.org/x/tools/cmd/guru": verifying non-authoritative meta tag
Fetching https://golang.org/x/tools?go-get=1
Parsing meta tags from https://golang.org/x/tools?go-get=1 (status code 200)
package golang.org/x/tools/cmd/guru: cannot download, $GOPATH not set. For more details see: go help gopath
Fetching https://golang.org/x/tools/cmd/guru?go-get=1
Parsing meta tags from https://golang.org/x/tools/cmd/guru?go-get=1 (status code 200)
get "golang.org/x/tools/cmd/guru": found meta tag main.metaImport{Prefix:"golang.org/x/tools", VCS:"git", RepoRoot:"https://go.googlesource.com/tools"} at https://golang.org/x/tools/cmd/guru?go-get=1
get "golang.org/x/tools/cmd/guru": verifying non-authoritative meta tag
Fetching https://golang.org/x/tools?go-get=1
Parsing meta tags from https://golang.org/x/tools?go-get=1 (status code 200)
package golang.org/x/tools/cmd/guru: cannot download, $GOPATH not set. For more details see: go help gopath

gorename:
Error: Command failed: /usr/local/bin/go get -u -v golang.org/x/tools/cmd/gorename
Fetching https://golang.org/x/tools/cmd/gorename?go-get=1
Parsing meta tags from https://golang.org/x/tools/cmd/gorename?go-get=1 (status code 200)
get "golang.org/x/tools/cmd/gorename": found meta tag main.metaImport{Prefix:"golang.org/x/tools", VCS:"git", RepoRoot:"https://go.googlesource.com/tools"} at https://golang.org/x/tools/cmd/gorename?go-get=1
get "golang.org/x/tools/cmd/gorename": verifying non-authoritative meta tag
Fetching https://golang.org/x/tools?go-get=1
Parsing meta tags from https://golang.org/x/tools?go-get=1 (status code 200)
package golang.org/x/tools/cmd/gorename: cannot download, $GOPATH not set. For more details see: go help gopath
Fetching https://golang.org/x/tools/cmd/gorename?go-get=1
Parsing meta tags from https://golang.org/x/tools/cmd/gorename?go-get=1 (status code 200)
get "golang.org/x/tools/cmd/gorename": found meta tag main.metaImport{Prefix:"golang.org/x/tools", VCS:"git", RepoRoot:"https://go.googlesource.com/tools"} at https://golang.org/x/tools/cmd/gorename?go-get=1
get "golang.org/x/tools/cmd/gorename": verifying non-authoritative meta tag
Fetching https://golang.org/x/tools?go-get=1
Parsing meta tags from https://golang.org/x/tools?go-get=1 (status code 200)
package golang.org/x/tools/cmd/gorename: cannot download, $GOPATH not set. For more details see: go help gopath

godef:
Error: Command failed: /usr/local/bin/go get -u -v github.com/rogpeppe/godef
package github.com/rogpeppe/godef: cannot download, $GOPATH not set. For more details see: go help gopath
package github.com/rogpeppe/godef: cannot download, $GOPATH not set. For more details see: go help gopath

golint:
Error: Command failed: /usr/local/bin/go get -u -v github.com/golang/lint/golint
package github.com/golang/lint/golint: cannot download, $GOPATH not set. For more details see: go help gopath
package github.com/golang/lint/golint: cannot download, $GOPATH not set. For more details see: go help gopath

gotests:
Error: Command failed: /usr/local/bin/go get -u -v github.com/cweill/gotests/...
package github.com/cweill/gotests/...: cannot download, $GOPATH not set. For more details see: go help gopath
package github.com/cweill/gotests/...: cannot download, $GOPATH not set. For more details see: go help gopath
```

**Comment by ramya-rao-a** [comment]
@amazingandyyy In your case the error message in the console says that you have not set GOPATH
You can set it using `go.gopath` setting or better yet, close all VS Code winodws, set GOPATH as an environment variable outside VS Code, and open VS Code again

**Comment by klingtnet** [comment]
I've a solution for Gnome users and probably everyone else who starts code by its `.desktop` file:

- the problem is that if you start `code` through the application launcher then it will not inherit the environment variables exported from your `.bashrc, .zshrc, .whateverrc`
- the integrated terminal of vscode will show the correct `GOPATH` but only because it sources your shell-rc
- the solution is to create a `~/.profile` which also exports your `GOPATH`
- this file is sourced by GNOME, and probably some other DE's, and makes the environment variables available for GUI applications
- the `~/.profile` could simply look like this:

```sh
# ~/.profile
export GOPATH="$HOME/.go"
```

**Comment by kevinmcfarlane** [comment]
I ran into this over the weekend. None of the above worked. This is on Win 10 Pro Anniversary Update with all patches. And latest version of VS Code as I write.

However, looking at the "install tools" message it was saying run `go get -v xxx`
But clicking on Install or Install All would run `go get -u -v xxx` and report fail even though it was creating all the folders and files.

So I went to the command prompt and typed `go get -v xxx` for one of the tools and it worked. So I then repeated for the remaining tools and now it's fine.

**Comment by lunemec** [comment]
After update this morning the same issue happened to me as well. It says `Cannot find "go" binary. Update PATH or GOROOT appropriately`. I'm using mac, and Go is insalled with homebrew, so it has non-standart path. I'm setting this path in my .zshrc, and it works inside the integrated console, but I can't find a way to set PATH inside vscode. GOPATH is set correctly (as it worked yesterday before the update).

Got any ideas how can I get it working?

Also, there seems to be a bug with tools, it says:
`The "gofmt" command is not available. Use "go get -v undefined" to install.`
The "undefined" seems strange for "gofmt" :)

Thank you!

**Comment by ramya-rao-a** [comment]
@lunemec Must be the case of the files of the old version of the Go extension still lingering around. There is a bug on this in the VS Code repo, but I can't seem to find it.

But am glad you are unblocked, Happy Coding!

**Comment by rakyll** [comment]
I came across to this issue, re-installing and reloading the window didn't work. But, uninstalling, quitting VSCode, and reinstalling did the trick.

**Comment by ramya-rao-a** [comment]
@rakyll Which version of VS Code are you on?

**Comment by rakyll** [comment]
Version 1.13.0 (1.13.0)

Let me explain further how this is reproducible.

I am on a fresh new laptop with no Go installed. I installed VSCode and the Go extension before Go. Then, I installed Go from source and add the GOROOT/bin to my PATH. Even though, my bash was able to recognize the go command, VSCode was prompting the error message saying that I don't have go in my PATH.

**Comment by ramya-rao-a** [comment]
@rakyll At that point VS Code is not aware of the addition of GOROOT/bin to the PATH. At this point, if you close VS Code and open it from the terminal where you updated the PATH, VS Code should now be able to find Go. Does it not?

**Comment by rakyll** [comment]
@ramya-rao-a, it didn't. I exited VS Code and opened it from the terminal where `go` is available for a few times before giving up.

**Comment by klingtnet** [comment]
@rakyll What system are you using, Linux, Mac or Windows? Can vscode's integrated terminal fing `go`?

**Comment by rakyll** [comment]
@klingtnet, I am on a Mac. The integrated terminal also couldn't find go.

**Comment by ramya-rao-a** [comment]
I'll try and get a repro and get back to this thread

**Comment by ZacharyZYH** [comment]
I was in the same trouble, and I found it was caused by the Git Path. 
My Git Path was in Drive C which the Code did not have the license to operate. After I reinstalled Git in Drive E, the problem was solved.

**Comment by aleafboat** [comment]
I was in the same trouble https fetch failed: Get https://golang.org/x/tools/imports?go-get=1: dial tcp 216.239.37.1:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

**Comment by fruser** [comment]
Having issues with running GO files from within Code. I've configured GOPATH but it didn't help. This is the error message I'm getting when I try to run the code:
```
[Running] go run "/Users/username/Projects/GO/src/hello/hello.go"
/bin/sh: go: command not found
```
Extension does work if I run it from within the terminal. Are there any other env variables I can configure in Settings file to make it work outside the terminal?
